src/main/java/com/iota/iri/BundleValidator.java

```diff
@@ -8,8 +8,40 @@
 
 import java.util.*;
 
+/**
+ * Validates bundles.
+ * <p>
+ * Bundles are a list of transactions that represent an atomic transfer, meaning that either all
+ * transactions inside the bundle will be accepted by the network, or none.
+ *</p>
+ */
 public class BundleValidator {
 
+    /**
+     * Fetches a bundle of transactions identified by the {@code tailHash} and validates them.
+     * <p>
+     * The fetched transactions have the same bundle hash as the transaction identified by {@code tailHash}
+     * The validation ascertains that:
+     * <ol>
+     *     <li>{@code tailHash} has an index of 0</li>
+     *     <li>{@code tailHash} was not already invalidated by a previous call to this method</li>
+     *     <li>That the transactions reference order is consistent with the indexes</li>
+     *     <li>The last index of each transaction in the bundle matches the last index of the tail transaction</li>
+     *     <li>While summing up the bundle value we never exceed the total supply of iotas</li>
+     *     <li>Check that last trit in a valid address hash is 0. We generate addresses using Kerl and we lose
+     *     the last trit when we convert from binary</li>
+     *     <li>Ascertain that total bundle value is 0 (inputs and outputs are balanced)</li>
+     *     <li>Recalculate the bundle hash by absorbing and squeezing the transaction essence of the transactions</li>
+     *     <li>Validate the signature on input transactions</li>
+     * </ol>
+     *</p>
+     * @param tangle used to fetch the bundle's transactions from the persistence layer
+     * @param tailHash the hash of the last transaction in a bundle.
+     * @return A list of transactions of the bundle contained in another list. If the bundle is valid then the tail transaction's
+     * {@link TransactionViewModel#getValidity()} will return 1, else it will return -1. If the bundle is invalid
```

Comment:
the `-1` part is confusing, it seems like the "will return 1, else it will return -1" is for this method, not the getValidity

Reply:
:+1: